src/ImageSharp/Formats/Webp/WebpAnimationDecoder.cs

```diff
@@ -0,0 +1,408 @@
+        /// <param name="frameX">The frame x coordinate.</param>
+        /// <param name="frameY">The frame y coordinate.</param>
+        /// <param name="frameWidth">The width of the frame.</param>
+        /// <param name="frameHeight">The height of the frame.</param>
+        private void DrawDecodedImageOnCanvas<TPixel>(Image<TPixel> decodedImage, ImageFrame<TPixel> imageFrame, int frameX, int frameY, int frameWidth, int frameHeight)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            Buffer2D<TPixel> decodedImagePixels = decodedImage.Frames.RootFrame.PixelBuffer;
+            Buffer2D<TPixel> imageFramePixels = imageFrame.PixelBuffer;
+            int decodedRowIdx = 0;
+            for (int y = frameY; y < frameY + frameHeight; y++)
+            {
+                Span<TPixel> framePixelRow = imageFramePixels.DangerousGetRowSpan(y);
+                Span<TPixel> decodedPixelRow = decodedImagePixels.DangerousGetRowSpan(decodedRowIdx++).Slice(0, frameWidth);
+                decodedPixelRow.TryCopyTo(framePixelRow.Slice(frameX));
+            }
+        }
+
+        /// <summary>
+        /// After disposing of the previous frame, render the current frame on the canvas using alpha-blending.
+        /// If the current frame does not have an alpha channel, assume alpha value of 255, effectively replacing the rectangle.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="src">The source image.</param>
+        /// <param name="dst">The destination image.</param>
+        /// <param name="frameX">The frame x coordinate.</param>
+        /// <param name="frameY">The frame y coordinate.</param>
+        /// <param name="frameWidth">The width of the frame.</param>
+        /// <param name="frameHeight">The height of the frame.</param>
+        private void AlphaBlend<TPixel>(ImageFrame<TPixel> src, ImageFrame<TPixel> dst, int frameX, int frameY, int frameWidth, int frameHeight)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            Buffer2D<TPixel> srcPixels = src.PixelBuffer;
+            Buffer2D<TPixel> dstPixels = dst.PixelBuffer;
+            Rgba32 srcRgba = default;
+            Rgba32 dstRgba = default;
+            PixelBlender<Rgba32> blender = PixelOperations<Rgba32>.Instance.GetPixelBlender(PixelColorBlendingMode.Normal, PixelAlphaCompositionMode.SrcOver);
+            for (int y = frameY; y < frameY + frameHeight; y++)
+            {
+                Span<TPixel> srcPixelRow = srcPixels.DangerousGetRowSpan(y);
+                Span<TPixel> dstPixelRow = dstPixels.DangerousGetRowSpan(y);
+                for (int x = frameX; x < frameX + frameWidth; x++)
+                {
+                    ref TPixel srcPixel = ref srcPixelRow[x];
+                    ref TPixel dstPixel = ref dstPixelRow[x];
+                    srcPixel.ToRgba32(ref srcRgba);
+                    dstPixel.ToRgba32(ref dstRgba);
```

Comment:
It would be better if we could convert the rows using bulk operations per scanline